src/components/MasterDetails.js

```diff
@@ -137,6 +137,31 @@ export default class MasterDetails extends React.Component<MasterDetailsDefaultP
   }
   state: MasterDetailsState;
 
+  componentWillReceiveProps(newProps: MasterDetailsProps) {
```

Comment:
[I should have remembered this earlier, but componentWillReceiveProps() is deprecated and will go away in React 17. I wouldn't change anything now... It's still okay to use it, but we'll eventually need to update this to be a different lifecycle method, like, for example, componentDidUpdate...]